src/Features/Core/Portable/EmbeddedLanguages/Classification/AbstractEmbeddedLanguageClassificationService.cs

```diff
@@ -106,17 +106,20 @@ public void VisitTokens(SyntaxNode node)
             private void ProcessToken(SyntaxToken token)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
-                ProcessTriviaList(token.LeadingTrivia);
+
+                // Directives need to be processes as they can contain strings, which then have escapes in them.
+                if (token.ContainsDirectives)
+                    ProcessTriviaList(token.LeadingTrivia);
+
                 ClassifyToken(token);
-                ProcessTriviaList(token.TrailingTrivia);
```

Comment:
we were also previous walking down into trailing trivia (which never has structure).  we were also walking into *all* structure, even though we don't have any embedded-lang classification in structured-trivia *except* for the case of `string literals` in directives.  so this means no more walking into things like doc comments, which is nice.